### PR TITLE
Another plugin: Titleregexp - modify titles of albums with regular expressions

### PR DESCRIPTION
**Pull request by tudacs**
This PR adds a plugin called 'titleregexp' which modifies titles of albums by configured regular expressions and simple search-and-replace tuples.

### CODE REVIEW

File: sigal/plugins/titleregexp.py [hunk 1]
```diff
@@ -0,0 +1,85 @@
...
+"""
+
+import logging
+import os
+import re
+
+from sigal import signals
+
+logger = logging.getLogger(__name__)
+cfg = {}
```
Comment (saimn) [inline sigal/plugins/titleregexp.py:52]: Define `cfg` here seems useless.

File: sigal/plugins/titleregexp.py [hunk 2]
```diff
@@ -0,0 +1,85 @@
...
+simple character replacements.
+
+Settings:
+
+- ``titleregexp`` with the following keys:
+    - ``regexp``, which is an array of dicts with 'search', 'replace' and 
+        'count' keys
+    - ``substitute``, which is an array of 2-element-arrays, of which the
+        second element denotes the replacement of occurences of the first 
+        element
```
Comment (saimn) [inline sigal/plugins/titleregexp.py:31]: Could you explain a bit more what the regex is doing ? (I see it's using `re.subn` but it's not clear from the docs)

File: sigal/plugins/titleregexp.py [hunk 3]
```diff
@@ -0,0 +1,85 @@
...
+            for s in r.get('substitute', []) :
+                album.title = album.title.replace(s[0],s[1])
+            if r.get('break','') != '' :
+                break
+
+    for r in cfg.get('substitute', []) :
+        album.title = album.title.replace(r[0],r[1])
+
+    if total > 0:
+        logger.info("Fixing title to '%s'", album.title)
```
Comment (saimn) [inline sigal/plugins/titleregexp.py:78]: Maybe log the original title as well, to make it clear which one has been changed.

**Review by saimn — COMMENTED** [review]
A few comments below. Also please add your plugin to `docs/plugins.rst`, thanks.

**Comment by tudacs** [comment]
Thank you for looking into this, your remarks are appreciated :). I added the documentation as suggested and tried to be as clear as possible. If the docs are too technical please let me know and I'll try to adapt it.

**Comment by saimn** [comment]
Looks good, thanks.

**Pull request merged**